skimage/feature/util.py

```diff
@@ -145,22 +145,32 @@ def plot_matched_features(
     ax.imshow(image, cmap='gray')
     ax.axis((0, image0.shape[1] + offset[1], image0.shape[0] + offset[0], 0))
 
-    rng = np.random.default_rng()
+    number_of_matches = matches.shape[0]
 
-    for i in range(matches.shape[0]):
-        idx0 = matches[i, 0]
-        idx1 = matches[i, 1]
+    from matplotlib.colors import is_color_like
 
-        if matches_color is None:
-            color = rng.random(3)
-        else:
-            color = matches_color
+    if matches_color is None:
+        rng = np.random.default_rng(seed=0)
+        colors = [rng.random(3) for _ in range(number_of_matches)]
+    elif is_color_like(matches_color):
+        colors = [matches_color for _ in range(number_of_matches)]
+    elif hasattr(matches_color, "__len__") and len(matches_color) == number_of_matches:
+        # No need to check, each color, matplotlib does so for us
+        colors = matches_color
+    else:
+        error_message = (
+            '`matches_color` needs to be a single color '
+            'or a sequence with a color for each match.'
```

Comment:
```suggestion
            'or a sequence of length equal to the number of matches.'
```
We might as well be explicit, because a common behaviour with colormaps is to cycle over all the colors... We could also imagine that it would be fine if the number of matches were a multiple of the number of colors.

Reply:
Good idea. It seems to me that it does not necessarily need to be a multiple. We just start looping the colors from the beginning if we reach the end.

Reply:
@michaelbratsch yes, I agree that would be the most intuitive (and generous) behaviour!

Reply:
I picked up the idea to cycle the colors.

@mkcor @lagru Please have a look at the latest commit.